includes/formFactory/ManageWikiFormFactoryBuilder.php

```diff
@@ -244,8 +248,11 @@ private static function buildDescriptorSettings(
 		$formDescriptor = [];
 
 		foreach ( $config->get( 'ManageWikiSettings' ) as $name => $set ) {
-			$add = ( $set['from'] == 'mediawiki' ) ||  in_array( $set['from'], $extList );
-			$disabled = ( $ceMW ) ? !( !$set['restricted'] || ( $set['restricted'] && $permissionManager->userHasRight( $context->getUser(), 'managewiki-restricted' ) ) ) : true;
+			$mwRequirements = isset( $set['requires'] ) ? ManageWikiRequirements::process( $set['requires'], $extList, false, $wiki ) : true;
+			$visible = isset( $set['requires']['visibility'] ) || isset( $set['requires']['extensions'] ) ? $mwRequirements : true;
+
+			$add = ( $set['from'] == 'mediawiki' && $visible ) || ( in_array( $set['from'], $extList ) && $visible );
```

Comment:
Two $visible checks aren't needed, $visible && ( OR ) is all that is need

Reply:
fixed